### ISSUE TITLE
Prombles about datasets

### ISSUE THREAD

**Issue by dpj135** [fleonce/ITER#1]
Hello, how do I work with these datasets to fit your program?  Your folder ’scripts‘ doesn't contain a file 'datasets/load_datasets.sh'.

**Comment by fleonce** [comment]
Hi, sorry about that, the files were ignored by the `.gitignore` and I missed to add them to version control.
By running `scripts/datasets/load_datasets.sh`, the script should download all necessary datasets (outside of ACE05, which you will have to provide by yourself for licensing reasons)

**Comment by dpj135** [comment]
Thank you very much！If I want to run your program, what dependencies do I need to install?

> Hi, sorry about that, the files were ignored by the `.gitignore` and I missed to add them to version control. By running `scripts/datasets/load_datasets.sh`, the script should download all necessary datasets (outside of ACE05, which you will have to provide by yourself for licensing reasons)

**Comment by fleonce** [comment]
Running the pip install for the repository (if you have already cloned, run `pip install .`) should install all required dependencies.
If you want to install them manually, take a look at the `pyproject.toml` file

**Comment by dpj135** [comment]
Thank you very much！For this code in **iter/modeling_iter2.py:line:701** , I am a little confused. Do you have not implemented nested entities yet?
```py
if nested_training:
     assert self.is_feature_extra_lr_class
     raise NotImplementedError
```

**Comment by fleonce** [comment]
@dpj135 we have, training was done using the horribly unreadable code in `modeling_iter.py`, I reworked the file and missed that bit. It should be there now, if you have anything else, feel free to ask.

**Comment by dpj135** [comment]
> @dpj135 we have, training was done using the horribly unreadable code in `modeling_iter.py`, I reworked the file and missed that bit. It should be there now, if you have anything else, feel free to ask.
```python
num_left_to_nest_depth_index = unsqueeze_and_expand_dims(
                # [B, N, nest_depth] -> [B, N, nest_depth, num_right, num_left, T]
                closest_left_indices,
                (3, 4, 5),
                (3, num_right), (4, num_left), (5, self.num_types),
            )
            rr_pair_flag = torch.gather(
                rr_pair_flag,
                dim=2,
                index=num_left_to_nest_depth_index
            )
            # [B, N, nest_depth, num_right, nest_depth, T]
            # here is the problem
            rr_pair_flag = torch.gather(
                rr_pair_flag,
                dim=4,
                index=unsqueeze_and_expand_dims(
                    is_right_closest_left_indices,
                    (1, 2, 5),
                    (1, seq_len), (2, nest_depth), (5, self.num_types)
                )
            )
```
Thank you！I find some problems with your code **in iter/modeling_iter2.py:line:780&795**. I think `self.num_types` may be `self.num_links`.
I found that there is not enough video memory during training. I use 3090  24GB. May I ask if you have any suggestions？

**Comment by fleonce** [comment]
You are right, it should have been `self.num_links` in both cases.
As for required GPU memory, I can suggest using --use_bfloat16 as a training argument or resorting to smaller models.

**Comment by dpj135** [comment]
Thank you. Your work is wonderful.

**Comment by dpj135** [comment]
Sorry to disturb you again.
  What are the loss functions of **is_left, is_span, and is_link**? It is not mentioned in the paper. 
  In the paper, the result of **is_left** should be a numerical value, but I found that the code made a connection between 'left' and 'action', and finally completed the classification through softmax. How to understand this 'action' and what does it mean in the original paper?

**Comment by fleonce** [comment]
Hi, sorry for responding so late, got caught up in university stuff.

The three loss functions are mentioned in the paper in Appendix B.
The result of is_left is not a numerical value, there is a criterion $is-left(x) > 0$ (Eq. 4), similar to the code

https://github.com/fleonce/ITER/blob/39b95b72b7414a58d6762363f0fe115452317db8/iter/modeling_iter2.py#L319

`action` is just a helper variable encoding which actions the model is performing at each token and can hold multiple values, depending on whether `is_left` and/or `is_span` are true.
